### PR TITLE
feat: Add dedicated method to load init scripts

### PR DESCRIPTION
**Pull request by juliusknorr**
During development on text I noticed that the new files2vue file list is loading too fast and fetching the file list before text has even had a chance to register additional dav properties on the initial page load.

This adds a method to allow apps registering init scripts which are loaded first (even before any core javascript). They need to be handled with care, but I indicated that also on the phpdoc comment.

Change itself is from @skjnldsv, I just shaped it to a pull request.

Example usage: https://github.com/nextcloud/text/pull/4776/commits/0eb23bbf8a3afc534a62e692957d00f4222a56ba

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: lib/public/Util.php [hunk 1]
```diff
@@ -163,6 +158,26 @@ public static function addStyle($application, $file = null) {
 		\OC_Util::addStyle($application, $file);
 	}
 
+	/**
+	 * Add a standalone init js file that is loaded for initialization
+	 *
+	 * Be careful loading scripts using this method as they are loaded early
+	 * and block the initial page rendering.
+	 * They should not have dependencies on any of the core javascript logic
+	 * as it is also not loaded at that stage.
```
Comment (skjnldsv) [inline lib/public/Util.php:167]: This is not really true, like you can require OC properties at this stage
It is purposely being pushed after the `core-common` file.

Comment (juliushaertl) [reply lib/public/Util.php:167]: Ah then my memory was wrong in that, will adjust

Comment (skjnldsv) [reply lib/public/Util.php:167]: image

**Review by skjnldsv — APPROVED** [review]

File: lib/public/Util.php (hunk 1 above)
Comment (juliushaertl) [reply lib/public/Util.php:167]: Adjusted

**Review by skjnldsv — APPROVED** [review]

**Review by come-nc — APPROVED** [review]

File: lib/public/Util.php [hunk 2]
```diff
@@ -214,6 +228,7 @@ public static function getScripts(): array {
 		$sortedScripts = $scriptSort->sort(self::$scripts, self::$scriptDeps);
 
 		// Flatten array and remove duplicates
+		$sortedScripts = array_merge([self::$scriptsInit], $sortedScripts);
 		$sortedScripts = $sortedScripts ? array_merge(...array_values(($sortedScripts))) : [];
```
Comment (come-nc) [inline lib/public/Util.php:232]:
```suggestion
		$sortedScripts = array_merge(...array_values($sortedScripts));
```

**Comment by ChristophWurst** [comment]
this new feature needs docs

**Pull request merged**